### ISSUE TITLE
Custoemr Credit made from a cash in is incorrect

### ISSUE THREAD

**Issue by joshxg** [openmsupply/mobile#2062]
## Describe the bug

Custoemr Credit made from a cash in is incorrect

### To reproduce

Dispensary development bug

### Expected behaviour

- Should have correct total, subtotal - the negative amount of the total of the RC

### Proposed Solution

N/A

### Version and device info

N/A

**Comment by joshxg** [comment]
cc should like like from a $30 receipt:

<img width="506" alt="image" src="https://user-images.githubusercontent.com/35858975/73413331-3272bd80-4370-11ea-9613-0ce3b0296191.png">


but currently looks like:

<img width="244" alt="image" src="https://user-images.githubusercontent.com/35858975/73413345-3dc5e900-4370-11ea-9d2d-bdc3bfcbcb3a.png">

**Comment by DhanyaHerath** [comment]
The same applies for Customer Invoices made from a cash out (totals are 0, should be the amount of the PS)

**Comment by wlthomson** [comment]
Looks like this is caused by `outgoingSync` for transactions. Currently, `total` and `subtotal` are set with `totalPrice`:

```
 get totalPrice() {
    return getTotal(this.items, 'totalPrice');
  }
```

Since `CashIn` and `CashOut` records don't create any `TransactionItem` records, that would explain this behaviour. 

Simplest solution to my mind would be to do a check for `record.type` and replace the call to `totalPrice` with `total` for records of type `payment` and `receipt`. Would this interfere with any dispensing payment logic?

**Comment by wlthomson** [comment]
Update: 

Looks like there might be two issues here. The first is with the `payment`/`receipt` records and the second with the customer invoices and supplier credits. For the former, the fix is described above (have replicated and tested fix). For the latter, when the customer invoices/supplier credits are generated, an associated `TransactionBatch` is created, but no associated `TransactionItem`. As `transaction.getTotal` accesses transaction batches via the transaction items, the total is always coming back as `0`. 

I think this can be fixed by updating the `createPaymentLine` and  `createReceiptLine` helpers. Will update with more info.

**Comment by joshxg** [comment]
Closed by #2136
